### PR TITLE
VZ Events: City_id

### PR DESCRIPTION
**Pull request by sergiogcx**
Closes https://github.com/cityofaustin/atd-data-tech/issues/3689

This is currently hooked up with VZD-Staging, to test on your end:
1. Find a crash that  is full purpose and has coordinates, copy the coordinates. And keep them in a safe notepad for later.
2. Find a crash that is not city_id 22 and is not full purpose, NOTE THE OLD CITY ID paste the coordinates and save.
3. Wait 10-20 seconds, see the magic happen (the city id will magically be updated to 22)

### CODE REVIEW

File: atd-events/crash_update_jurisdiction/app.py [hunk 1]
```diff
@@ -67,6 +69,46 @@ def get_jurisdiction_flag(data: dict) -> str:
         return "N"
 
 
+def get_city_id(data: dict) -> Optional[int]:
```
Comment (mddilley) [inline atd-events/crash_update_jurisdiction/app.py:72]: Cool annotations here! I haven't seen this syntax before so it was fun looking into it. Is there any type checking here or is it for documentation? I read conflicting things so I thought that I would just ask. 👍

File: atd-events/crash_update_jurisdiction/app.py [hunk 2]
```diff
@@ -67,6 +69,46 @@ def get_jurisdiction_flag(data: dict) -> str:
...
+    """
+    Compares two arrays of integers, and returns a set of unique common members
+    :param List[int] a: An array you want to compare against
+    :param List[int] b: The other array you want to compare against
+    :return Set[int]:
+    """
+    a_set = set(a)
+    b_set = set(b)
+
+    if a_set & b_set:
```
Comment (mddilley) [inline atd-events/crash_update_jurisdiction/app.py:106]: I didn't realize that you could compare sets like this! Very handy way to find the common elements. 😎

File: atd-events/crash_update_jurisdiction/app.py [hunk 3]
```diff
@@ -83,6 +125,45 @@ def load_data(record: str) -> dict:
...
+                    "query": find_crash_jurisdictions,
+                    "variables": {
+                        "crash_id": crash_id
+                    }
+                }
+            ),
+            headers=HEADERS
+        )
+        valid_jurisdictions = [5, 3, 7, 8, 10]
+        output_jurisdictions = list(map(lambda x: x['id'], response.json()["data"]["find_crash_jurisdictions"]))
```
Comment (mddilley) [inline atd-events/crash_update_jurisdiction/app.py:159]: Cool lambda here! Gives me JS vibes, and I like it. 😄

**Review by mddilley — APPROVED** [review]
I tested as directed and saw the city ID update just as expected. 🙌  Thanks for listing out the steps to test it. Lots of learning here 🐍, and your tests helped me see how everything fits together - good stuff!

File: atd-events/crash_update_jurisdiction/app.py (hunk 1 above)
Comment (sergiogcx) [reply atd-events/crash_update_jurisdiction/app.py:72]: There is no type checking, only syntax. The only reason of these annotations is to help the developers understand what they are working with.

File: atd-events/crash_update_jurisdiction/app.py (hunk 2 above)
Comment (sergiogcx) [reply atd-events/crash_update_jurisdiction/app.py:106]: Yea, I had not been using sets before and I think I need to make more use of them going forward. They are a swiss-army knive for data processing

File: atd-events/crash_update_jurisdiction/app.py (hunk 1 above)
Comment (mddilley) [reply atd-events/crash_update_jurisdiction/app.py:72]: Awesome, thanks! They are very helpful. 👍

**Pull request merged**